File: report_item.cpp
#include <iostream>
#include <string>
#include <mysql_driver.h>
#include <mysql_connection.h>
#include <cppconn/prepared_statement.h>
#include <cppconn/resultset.h>

using namespace std;

int main() 
{
    
    cout << "Content-type: text/html\n\n";

    // get form data 
    string user_id = getenv("QUERY_STRING");  
    string item_name = getenv("ITEM_NAME");
    string description = getenv("DESCRIPTION");
    string lost_date = getenv("LOST_DATE");
    string location = getenv("LOCATION");

    // check if the values are retrieved
    if (item_name.empty() || description.empty() || lost_date.empty() || location.empty()) {
        cout << "<html><body><h3>Error: All fields must be filled out!</h3></body></html>";
        return 1;
    }

    try 
      {
        // connection to the MySQL database
        sql::mysql::MySQL_Driver *driver;
        sql::Connection *con;

        driver = sql::mysql::get_mysql_driver_instance();
        con = driver->connect("tcp://127.0.0.1:3306", "root", "root"); 
        con->setSchema("lost_found");  //database

        // INSERT SQL statement
        sql::PreparedStatement *pstmt;
        pstmt = con->prepareStatement("INSERT INTO lost_items (user_id, item_name, description, lost_date, location) VALUES (?, ?, ?, ?, ?)");

        // query parameters
        pstmt->setInt(1, stoi(user_id));  
        pstmt->setString(2, item_name);
        pstmt->setString(3, description);
        pstmt->setString(4, lost_date);
        pstmt->setString(5, location);

        // execute query
        pstmt->executeUpdate();

        // Close the statement and connection
        delete pstmt;
        delete con;

        // message to the user
        cout << "<html><body><h3>Item Reported Successfully!</h3></body></html>";
      } catch (sql::SQLException &e) 
        {
        //  return an error message
          cout << "<html><body><h3>Error: " << e.what() << "</h3></body></html>";
          return 1;
        }

    return 0;
}

File: report_item.php
<?php
//db connection
$con = new mysqli("localhost", "root", "Sp0ngebob41913!?", "lost_found");

//check connection
if ($con->connect_error) {
    die("Connection failed: " . $con->connect_error);
}

//get form data
$item_name = $_POST['item_name'];
$description = $_POST['description'];
$lost_date = $_POST['lost_date'];
$location = $_POST['location'];
$email = $_POST['email'];

//image upload
$target_dir = "uploads/";
$image_name = basename($_FILES["image"]["name"]);
$target_file = $target_dir . time() . "_" . $image_name;
$imageFileType = strtolower(pathinfo($target_file, PATHINFO_EXTENSION));

//validate image 
$check = getimagesize($_FILES["image"]["tmp_name"]);
if ($check === false) {
    die("File is not an image.");
}


if (!move_uploaded_file($_FILES["image"]["tmp_name"], $target_file)) {
    die("Sorry, there was an error uploading your file.");
}

//insert user contact info into users table
$stmt = $con->prepare("INSERT INTO users (name, email) VALUES (?, ?)");
$stmt->bind_param("ss", $item_name, $contact);
$stmt->execute();
$user_id = $stmt->insert_id;
$stmt->close();

//insert item into lost_items table
$stmt = $con->prepare("INSERT INTO lost_items (user_id, item_name, description, lost_date, location, image_path) VALUES (?, ?, ?, ?, ?, ?)");
$stmt->bind_param("isssss", $user_id, $item_name, $description, $lost_date, $location, $target_file);
$stmt->execute();
$stmt->close();

echo "Item reported successfully.";
$con->close();
?>

